catroid/src/main/java/org/catrobat/catroid/ui/FormulaEditorCopyPasteController.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2023 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui
+
+import android.annotation.SuppressLint
+import androidx.annotation.VisibleForTesting
+import org.catrobat.catroid.formulaeditor.FormulaEditorEditText
+import org.catrobat.catroid.formulaeditor.InternToken
+
+import java.util.ArrayList
+
+@SuppressLint("InflateParams")
+class FormulaEditorCopyPasteController(private val formulaEditorEditText: FormulaEditorEditText) {
+
+    @VisibleForTesting
+    var clipboard: List<InternToken>? = null
```

Comment:
We also need to consider whether to close https://jira.catrob.at/browse/IDE-34 because it is obsoleted by your changes.

This also gives us the need to consider whether the clipboard will need to be stored across app reboots, like the current implementation for IDE-34 does;

In my opinion, we should just leave it in-memory for the time being (that is how it was handled until now, after all) and consider whether we can move this to the system clipboard in a separate ticket.

It's probably best to discuss this in the meeting this evening.

Reply:
It is not obsolete, since I only copied the current clipboard behavior. 
The whole persistence stuff is still needed. And most of the changes done in IDE-34 are out of scope for this ticket.

Let's discuss it in the meeting.